### DISCUSSION TITLE
filter multiple key value pairs in JSON column in postgres

### DISCUSSION THREAD

**Discussion by sairam1318** [prisma/prisma#20089] [Q&A] [answered]
Hello, 
I have a use case where I have to query a Json column which matches multiple key value pairs. Does Prisma support this for Postgres. I know one way it to use path and equals and but the object I receive was a bit huge and is dynamic. Help would be appreciated. Sorry If this was already solved by someone in the discussions.

Example JSON: 
```json
{
    "name": "eyra",
    "age": "F",
    "course": "MSC",
    "native": "India",
    "fatherName": "Satya",
    "education": "B.Tech"
}
```
Example filter:
```json
{
    "name": "eyra",
    "native": "India"
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @sairam1318 👋 

You can achieve this using this query
```js
const getUsers = await prisma.user.findMany({
  where: {
    AND: [
      {
        userInfo: {
          path: ['name'],
          equals: 'eyra',
        },
      },
      {
        userInfo: {
          path: ['native'],
          equals: 'India',
        },
      },
    ],
  },
})
```
You can learn more by reading this [section](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-nested-object-property) of the documentation.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by sairam1318** [reply]
Yeah great, but I see `NOT` is not working as expected. Can you also guide me the solution here.

**Reply by janpio** [reply]
What exactly is not working as expected around `NOT`?

**Reply by GayathriAmarneni123** [reply]
In the above query we are finding users that matches the userInfo. In the same way I need user data that not matches the userInfo.
Probably we use not in that case. But here in Json fields it is giving empty data if I use NOT.

**Reply by janpio** [reply]
Can you share the query that is not working?
